modules/tensor_mechanics/include/materials/StressUpdateBase.h

```diff
@@ -157,6 +157,11 @@ class StressUpdateBaseTempl : public Material
    */
   virtual void storeIncrementalMaterialProperties(){};
 
+  /**
+   * Zero out material properties. Useful for substepping with inelastic models.
+   */
+  virtual void zeroOutIncrementalMaterialProperties(){};
```

Comment:
If this is called, does it mean the user is trying to do sub-stepping but the model does not support it? If so, maybe we should throw an error or something?

Reply:
No, you can always call this method. It may or may not be implemented downstream, but that's responsibility of the model. I do think we are not handling an error when the user selects `use_substep = true` and the model doesn't support it (it does not inherit that flag). I'll look into creating that paramError.